### PR TITLE
feat: add back action creator and helper

### PR DESCRIPTION
**Pull request by erik-slovak**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "@react-navigation/devtools": "^6.0.10",
     "@react-navigation/native": "^6.0.13",
     "@txo-peer-dep/log": "^4.0.0",
+    "@txo-peer-dep/redux": "^1.0.0",
```
Comment (rostislav-simonik) [inline package.json:45]: Should not be here or ?

File: src/Navigation/Back.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    }
+  }
+  return getBackedState(
+    state,
+    typeof numberOfBacks === 'number' ? numberOfBacks - 1 : numberOfBacks,
+    destinationKey,
+  )
+}
+
+export const onBackAction = ({
```
Comment (rostislav-simonik) [inline src/Navigation/Back.ts:60]: if on action contain number of backs to be executed just execute back action multiple times without count param.

File: src/Model/Types.ts [hunk 3]
```diff
@@ -20,10 +20,14 @@ export type NavigationAction = RNNavigationAction & {
...
     params?: Record<string, unknown>,
   },
+  // requireConditions
+  conditionList?: Condition[],
   // navigate
   flow?: boolean,
   reset?: boolean,
   skipConditionalNavigation?: boolean,
+  // back
+  count?: number,
```
Comment (rostislav-simonik) [inline src/Model/Types.ts:30]: let's type for navigation action so we accidentally won't use parameters that are not related to particular type

```
export type NavigationAction = {
   type: 'back'
   count?: number
} | {
   type: 'navigate'
   flow?: boolean
} 
```

File: src/Navigation/Back.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+} from '../Model/Types'
+
+const log = new Log('txo.react-conditional-navigation.Navigation.Back')
+
+export const onBackAction = ({
+  action,
+  originalOnAction,
+  restArgs,
+}: OnActionAttributes): boolean => {
+  // TODO: add backToRouteName, key and routeName when necessary
```
Comment (rostislav-simonik) [inline src/Navigation/Back.ts:20]: Is this forgotten or ?

File: src/Navigation/Navigate.ts [hunk 5]
```diff
@@ -36,6 +36,9 @@ export const onNavigateAction = ({
   screenConditionConfigMap,
   setState,
 }: OnActionAttributes): boolean => {
+  if (action.type !== 'NAVIGATE') {
```
Comment (rostislav-simonik) [inline src/Navigation/Navigate.ts:39]: why is this necessary?

Comment (erik-slovak) [reply src/Navigation/Navigate.ts:39]: TypeScript shows an error that navigate specific attributes (`payload`, `flow`, `reset`, `skipConditionalNavigation`) are missing.

`return false` means that action was not handled, but this should never happen because the function is only called when `type === 'NAVIGATE'`, I just needed to tell TypeScript that.

File: src/Navigation/Back.ts (hunk 4 above)
Comment (erik-slovak) [reply src/Navigation/Back.ts:20]: It's not forgotten, but it's unnecessary, removed.

File: src/Navigation/Navigate.ts [hunk 6]
```diff
@@ -36,6 +36,9 @@ export const onNavigateAction = ({
   screenConditionConfigMap,
   setState,
 }: OnActionAttributes): boolean => {
```
Comment (erik-slovak) [inline src/Navigation/Navigate.ts:38]:
```suggestion
}: OnActionAttributes<NavigateNavigationAction>): boolean => {
```

**Review by rostislav-simonik — APPROVED** [review]
